query-service-impl/src/test/java/org/hypertrace/core/query/service/attribubteexpression/AttributeExpressionSubpathExistsFilteringTransformationTest.java

```diff
@@ -0,0 +1,235 @@
+            .build();
+    Filter containsKeyFilter = createContainsKeyFilter("Span.tags", "span.kind");
+
+    QueryRequest originalRequest =
+        QueryRequest.newBuilder()
+            .setFilter(filter)
+            .addOrderBy(createOrderByExpression(spanTag, SortOrder.ASC))
+            .build();
+
+    QueryRequest expectedTransform =
+        QueryRequest.newBuilder()
+            .setFilter(
+                createCompositeFilter(
+                    Operator.AND,
+                    createCompositeFilter(Operator.AND, filter, containsKeyFilter).build(),
+                    containsKeyFilter))
+            .addOrderBy(createOrderByExpression(spanTag, SortOrder.ASC))
+            .build();
+
+    assertEquals(
+        expectedTransform,
+        this.transformation.transform(originalRequest, mockTransformationContext).blockingGet());
+  }
+
+  @Test
+  void transQueryWithComplexAttributeExpression_SingleSelection() {
+    Expression.Builder spanTag = createComplexAttributeExpression("Span.tags", "span.kind");
+
+    QueryRequest originalRequest = QueryRequest.newBuilder().addSelection(spanTag).build();
+
+    QueryRequest expectedTransform =
+        QueryRequest.newBuilder()
+            .addSelection(spanTag)
+            .setFilter(createContainsKeyFilter("Span.tags", "span.kind"))
+            .build();
+
+    assertEquals(
+        expectedTransform,
+        this.transformation.transform(originalRequest, mockTransformationContext).blockingGet());
+  }
+
+  @Test
+  void transQueryWithComplexAttributeExpression_SingleOrderBy() {
+    Expression.Builder spanTag = createComplexAttributeExpression("Span.tags", "span.kind");
+
+    QueryRequest originalRequest =
+        QueryRequest.newBuilder()
+            .addOrderBy(createOrderByExpression(spanTag, SortOrder.ASC))
+            .build();
+
+    QueryRequest expectedTransform =
+        QueryRequest.newBuilder()
+            .setFilter(createContainsKeyFilter("Span.tags", "span.kind"))
+            .addOrderBy(createOrderByExpression(spanTag, SortOrder.ASC))
+            .build();
+
+    assertEquals(
+        expectedTransform,
+        this.transformation.transform(originalRequest, mockTransformationContext).blockingGet());
+  }
+
+  @Test
+  void transQueryWithComplexAttributeExpression_MultipleOrderBy() {
+    Expression.Builder spanTag1 = createComplexAttributeExpression("Span.tags", "span.kind");
+    Expression.Builder spanTag2 = createComplexAttributeExpression("Span.tags", "FLAGS");
+
+    QueryRequest originalRequest =
+        QueryRequest.newBuilder()
+            .addOrderBy(createOrderByExpression(spanTag1, SortOrder.ASC))
+            .addOrderBy(createOrderByExpression(spanTag2, SortOrder.ASC))
+            .build();
+
+    QueryRequest expectedTransform =
+        QueryRequest.newBuilder()
+            .setFilter(
+                createCompositeFilter(
+                    Operator.AND,
+                    createContainsKeyFilter("Span.tags", "span.kind"),
```

Comment:
Since w are using a set now, is the order of filters ensured? Like sometimes this test may pass and sometimes it may fail? Similarly for other tests.

Reply:
It's an ordered set, there's no non-determinism (if you see any, please flag it - ImmutableSet or LinkedHashSet at least both maintain order)..